subsys/bluetooth/audio/broadcast_source.c

```diff
@@ -229,14 +228,20 @@ static int bt_audio_broadcast_source_setup_stream(uint8_t index,
 		return -ENOMEM;
 	}
 
+	iso = bt_audio_iso_new();
+	if (iso == NULL) {
+		BT_DBG("Could not allocate iso");
+		return -ENOMEM;
+	}
+
+	bt_audio_iso_init(iso, &broadcast_source_iso_ops);
+	bt_audio_iso_bind_ep(iso, ep);
+	bt_audio_iso_unref(iso);
+
 	bt_audio_stream_attach(NULL, stream, ep, codec);
-	ep->iso->source_stream = stream;
 	stream->qos = qos;
-	stream->iso->qos->tx->path = &ep->iso->source_path;
-	stream->iso->qos->tx->path->cc = ep->iso->source_path_cc;
-	stream->iso->qos->rx = NULL;
-	bt_audio_codec_qos_to_iso_qos(stream->iso->qos->tx, qos);
-	bt_audio_codec_to_iso_path(stream->iso->qos->tx->path, codec);
+	bt_audio_codec_qos_to_iso_qos(iso->chan.qos->tx, qos);
```

Comment:
iso is used after being unreffed

Reply:
OK, I agree it looks odd, I'll fix it. The point is that `ep` still holds the reference. See my comment here https://github.com/zephyrproject-rtos/zephyr/pull/51513/files/8c16dd8d7044627cce12f1f02a010c97c8b93fd5#r1012635325